Урок 6, ДЗ

vlastroG opened the pull request.

vlastroG committed:
Create L6.txt

vlastroG committed:
Удаление хлама

vlastroG committed:
№1

vlastroG committed:
№2

Infarh commented on BasicClasses/BankAccount.cs:
```diff
@@ -128,5 +128,30 @@ public BankAccount(BankAccountType type)
+
+        public static bool operator ==(BankAccount acc1, BankAccount acc2)
+        {
+            return (acc1.Type == acc2.Type);
```
Затлевший в acc1, или acc2 null убьёт вашу программу!
```suggestion
            return Equals(acc1, acc2);
```
Сравнение должно проводиться по номеру счёта, а не по типу!

Infarh commented on BasicClasses/BankAccount.cs:
```diff
@@ -128,5 +128,30 @@ public BankAccount(BankAccountType type)
+
+        public static bool operator !=(BankAccount acc1, BankAccount acc2)
+        {
+            return !(acc1.Type == acc2.Type);
```
```suggestion
            return !(acc1== acc2);
```

Infarh commented on BasicClasses/BankAccount.cs:
```diff
@@ -128,5 +128,30 @@ public BankAccount(BankAccountType type)
+
+        public bool Equals(BankAccount acc2)
+        {
+            return (this.Type == acc2.Type);
```
А если acc2 будет null? То Вас уволят с работы.
```suggestion
            return !ReferenceEquals(null, acc2) && Number == acc2.Number;
```

Infarh commented on BasicClasses/GeometryClasses/Circle.cs:
```diff
@@ -0,0 +1,44 @@
+        /// <returns>Площадь</returns>
+        public double Square()
+        {
+            return 3.14159265 * Radius * Radius;
```
```suggestion
            return Math.PI * Radius * Radius;
```
Фу, как грубо!

Infarh commented on BasicClasses/GeometryClasses/Rectangle.cs:
```diff
@@ -0,0 +1,47 @@
+        /// <param name="col">Цвет</param>
+        /// <param name="w">Ширина</param>
+        /// <param name="l">Длина</param>
+        public Rectangle(uint x, uint y, bool vis, Color col, uint w, uint l)
+            : base(x, y, vis, col) { Width = w; Length = l; }
```
```suggestion
        public Rectangle(uint x, uint y, bool vis, Color col, uint w, uint l)
            : base(x, y, vis, col) 
        {
            Width = w; 
            Length = l; 
        }
```

Infarh replied:
Имена переменных из одной буквы - плохие имена.

vlastroG committed:
Update BasicClasses/BankAccount.cs

Изменение сравнения банковских аккаунтов на проверку равенства их номеров с проверкой на NULL

Co-authored-by: Pavel <shmachilin@gmail.com>

vlastroG committed:
Update BasicClasses/GeometryClasses/Circle.cs

Math.PI вместо int цифры

Co-authored-by: Pavel <shmachilin@gmail.com>

vlastroG committed:
Update BasicClasses/GeometryClasses/Rectangle.cs

Форматирование кода. Замечание: надо давать переменным развернутые имена, а не из одной буквы

Co-authored-by: Pavel <shmachilin@gmail.com>

vlastroG committed:
Update BasicClasses/BankAccount.cs

Изменение проверки на неравенство банковских аккаунтов

Co-authored-by: Pavel <shmachilin@gmail.com>

vlastroG committed:
Update BasicClasses/BankAccount.cs

Изменение проверки на равенство двух банковских аккаунтов

Co-authored-by: Pavel <shmachilin@gmail.com>
